### ISSUE TITLE
After deleting account, it keeps the user logged in.

### ISSUE THREAD

**Issue by slushnys** [thepracticaldev/dev.to#6139]
After deleting an account permanently, I was still logged in with a user icon on the right side with options to view dashboard, sign out, etc. 

I had to manually sign out in order to access my other account.

**Expected behavior**

After deleting an account app should direct me to homepage without any user saved in the session.

**Comment by rhymes** [comment]
Thanks for the issue! We'll take your request into consideration and follow up if we decide to tackle this issue.

To our amazing contributors: issues labeled `type: bug` are always up for grabs, but for feature requests, please wait until we add a `ready for dev` before starting to work on it.

To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @thepracticaldev/oss and we will follow up within 3 business days.

For full info on how to contribute, please check out our [contributors guide](https://github.com/thepracticaldev/dev.to/blob/master/CONTRIBUTING.md).

**Comment by abdellani** [comment]
Hi
I want to take this one :)

**Comment by rhymes** [comment]
Go ahead @abdellani, and thank you!

**Comment by abdellani** [comment]
When I deleted an account using the console, the user found himself logged out after 1 or 2 interactions.
When I tried to delete the account via the dashboard (`/settings/account`), the application sends an email that contains a link. After clicking that link, the account was still present and fully functional. The only changes that I noticed in the admin board were in  **Opened_at** and **Clicked At** in `/admin/email_messages`. 
![Screenshot_2020-02-27 Email Messages - Practical Developer](https://user-images.githubusercontent.com/16113997/75443005-d5fbc180-5960-11ea-87b4-759288326061.png)

@rhymes
Does the user have any way to delete his account directly without waiting for an approval from an admin?
@slushnys 
Can you please provide me more details? did you just click in the link in `Account Deletion Requested` email? how much time did you wait after clicking that link? did you receive a message confirming that your account has been deleted?

**Comment by rhymes** [comment]
@rhymes yeah, the self deletion here https://dev.to/settings/account doesn't require an approval, does it? I thought one could just delete themselves by clicking on the link received in the email

**Comment by abdellani** [comment]
@rhymes 
I copied the log after clicking that link, and for me, it doesn't seem trying to remove the account 
https://gist.github.com/abdellani/09036b589cf83bcb154f79aa06aec4b9

**Comment by slushnys** [comment]
@abdellani I have clicked on the link and it kind of deleted the account as I lost information from in, and the status bar of app contained no user, but it just doesn't log out automatically.

**Comment by lightalloy** [comment]
@abdellani Could you describe what happens after you clicked the link in the email? It shouldn't delete the account right away after clicking the link, but lead to confirmation page, and the account should only be deleted after the confirmation.

**Comment by abdellani** [comment]
@lightalloy 
After clicking the link, The browser redirects me to the home page. I shared the output of the console in my previous comment.

**Comment by lightalloy** [comment]
@abdellani one possible problem could be that the link has expired, unfortunately, the log doesn't contain information about it (should include `UsersController#confirm_destroy`). 
One option is to notify a user in this case (e.g. add flash notice) to distinguish this case from other possible problems.

**Comment by mstruve** [comment]
Additional context from @juliannatetreault 
> I will say though, that the fact that the user can still see their profile picture, etc. is misleading -- if you attempt to click on your profile after deletion, a "Not Found" error will be raised, preventing the user from accessing their account as a deleted user. An ideal solution here would be to sign out the user and alert them that their account has been deleted and then redirect the deleted user to the sign in/sign up page. Currently, the user remains logged in and we redirect to the homepage (like the users said in the issue you linked to!), while simultaneously sending an email stating that the user's account was successfully deleted, but this current flow seems to muddy the deletion process and confirmation and I think we could definitely make some improvements here!
